app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx

```diff
@@ -6,30 +6,29 @@ import { Button, Pane } from "neetoui";
 import { Input, Select } from "neetoui/formik";
 
 import { CONTACTS, TAGS } from "components/Dashboard/constants";
-import { buildSelectOption } from "utils/index";
+import { buildSelectOption, parseNoteValues } from "utils/index";
 
 import { NOTES_FORM_VALIDATION_SCHEMA } from "../constants";
 
-const tags = TAGS.map(buildSelectOption);
-const contacts = CONTACTS.map(contact =>
+const TAG_OPTIONS = TAGS.map(buildSelectOption);
+const CONTACT_OPTIONS = CONTACTS.map(contact =>
   buildSelectOption(`${contact.firstName} ${contact.lastName}`)
 );
 
-const parseValues = values => {
-  const assignedContact = values.assignedContact.label;
-  const tags = values.tags.map(tag => tag.label);
-
-  return { ...values, tags, assignedContact };
-};
-
-const NoteForm = ({ onClose, note, isEdit, handleNote }) => {
+const NoteForm = ({
+  onClose,
+  note,
+  isEdit,
+  createNote = null,
+  editNote = null,
+}) => {
   const [submitted, setSubmitted] = useState(false);
 
   const handleSubmit = values => {
     try {
       isEdit
-        ? handleNote(note.id, parseValues(values))
-        : handleNote(parseValues(values));
+        ? editNote(note.id, parseNoteValues(values))
```

Comment:
We shouldn't use the `edit` keyword here. I believe you are trying to `update` the entity. `edit` means we only open up the form to edit something. Here we are `updating` in the backend from my understanding.